File: app/api/schemas/skills.py
# app/api/schemas/skills.py

from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, Field


class SkillSearchResponse(BaseModel):
    """Response model for a single skill in search results."""

    uri: str = Field(..., description="Skill URI")
    label: str = Field(..., description="Skill preferred label")
    description: Optional[str] = Field(None, description="Skill description")
    skillType: Optional[str] = Field(None, description="Skill type (skill/competence or knowledge)")

File: app/frontend/services/skills.ts
import api from '@/lib/api';

export interface Skill {
    uri: string;
    label: string;
    description?: string;
    skillType?: string;
}

export interface SkillSearchParams {
    q?: string;
    type?: string;  // 'knowledge' or 'skill/competence'
    groups?: string;  // comma-separated URIs
    schemes?: string;  // comma-separated URIs
    relatedTo?: string;  // skill URI
    limit?: number;
    offset?: number;
}

export const skillsService = {
    search: async (params: SkillSearchParams): Promise<Skill[]> => {
        const response = await api.get<Skill[]>('/skills', { params });
        return response.data;
    }
};
